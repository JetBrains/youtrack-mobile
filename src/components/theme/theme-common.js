/* @flow */

import type {UIThemeColorsPaletteColors} from '../../flow/Theme';

export const themeColorsPalette: UIThemeColorsPaletteColors = {
  $greyBackground: 'rgba(184, 209, 229, 0.15)',
  $greyColor: '#b8d1e5',

  $redBackground: 'rgba(219, 88, 96, 0.15)',
  $redColor: '#db5860',

  $yellowBackground: 'rgba(237, 162, 0, 0.15)',
  $yellowColor: '#eda200',

  $greenBackground: 'rgba(89, 168, 105, 0.15)',
  $greenColor: '#59a869',

  $blueBackground: 'rgba(0, 142, 255, 0.15)',
  $blueColor: '#008eff',
};
