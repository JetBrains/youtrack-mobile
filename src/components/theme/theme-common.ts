import type {UIThemeColorsPaletteColors} from 'types/Theme';
export const themeColorsPalette: UIThemeColorsPaletteColors = {
  $greyBackground: 'rgba(184, 209, 229, 0.15)',
  $greyColor: 'rgba(59, 107, 137, 0.35)',
  $redBackground: 'rgba(219, 88, 96, 0.15)',
  $redColor: '#db5860',
  $yellowBackground: 'rgba(237, 162, 0, 0.15)',
  $yellowColor: '#eda200',
  $greenBackground: 'rgba(89, 168, 105, 0.15)',
  $greenColor: '#59a869',
  $blueBackground: 'rgba(0, 142, 255, 0.15)',
  $blueColor: '#008eff',
};
