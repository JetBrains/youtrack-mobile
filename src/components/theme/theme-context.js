/* @flow */

import React from 'react';

export const ThemeContext = React.createContext<any>(null);
