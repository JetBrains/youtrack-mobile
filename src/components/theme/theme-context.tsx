import React from 'react';
export const ThemeContext: React.Context<any> = React.createContext<any>(null);
