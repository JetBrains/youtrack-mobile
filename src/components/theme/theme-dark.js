/* @flow */

import type {UITheme} from '../../flow/Theme';

const dark: UITheme = {
  dark: true,
  mode: 'ytdark',
  name: 'dark',
  barStyle: 'light-content',
  androidSummaryFontWeight: '400',

  colors: {
    $background: '#1c1c1e',
    $boxBackground: '#2c2c2e',

    $error: '#ee0e0e',

    $text: '#e1e1e1',
    $textSecondary: '#b0b0b0',
    $textButton: '#FFFFFF',

    $link: '#FF008C',
    $linkLight: 'rgba(254, 0, 130, 0.3)',

    $disabled: '#5d5d5d',

    $icon: '#9f9f9f',
    $iconAccent: '#b8d1e5',

    $mask: '#FFFFFF57',
    $navigation: '#696969',

    $separator: 'rgba(210, 254, 254, 0.15)'
  }
};

export default dark;
