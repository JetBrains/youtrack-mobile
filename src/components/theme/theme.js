/* @flow */

import {Appearance} from 'react-native-appearance';
import EStyleSheet from 'react-native-extended-stylesheet';

import lightTheme from './theme-light';
import darkTheme from './theme-dark';

import type {UITheme} from '../../flow/Theme';

export const DEFAULT_THEME: UITheme = lightTheme;
export const DEFAULT_MODE: string = lightTheme.name;

export const getSystemMode = (): string => Appearance.getColorScheme() || DEFAULT_MODE;
export const DEFAULT_SYSTEM_MODE: string = getSystemMode();

export const getUITheme = (themeName: string): UITheme => {
  if ([lightTheme.name, darkTheme.name].includes(themeName)) {
    return themeName === DEFAULT_MODE ? lightTheme : darkTheme;
  }
  return DEFAULT_THEME;
};

export const buildStyles = (mode: string = DEFAULT_SYSTEM_MODE, uiTheme: UITheme = DEFAULT_THEME) => {
  EStyleSheet.build({
    $theme: mode,
    $resolved: uiTheme.colors.$icon,
    $shadowColor: uiTheme.colors.$icon,
    $link: uiTheme.colors.$link,
    $iconAccent: uiTheme.colors.$iconAccent,
    $background: uiTheme.colors.$background,
    ...uiTheme.colors
  });
};

