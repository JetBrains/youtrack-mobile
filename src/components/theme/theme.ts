import {Appearance} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import AsyncStorage from '@react-native-async-storage/async-storage';
import darkTheme from './theme-dark';
import lightTheme from './theme-light';
import {THEME_MODE_KEY} from '../storage/storage';
import type {UITheme} from 'types/Theme';
export const DEFAULT_THEME: UITheme = lightTheme;
export const getSystemThemeMode = (): any => Appearance.getColorScheme();
export const themes: UITheme[] = [lightTheme, darkTheme];
export const getUITheme = (mode: string): UITheme => {
  const theme: UITheme | null | undefined = themes.reduce(
    (theme: UITheme | null | undefined, it: UITheme) =>
      it.mode.includes(mode) ? it : theme,
    null,
  );
  return theme || DEFAULT_THEME;
};
export const buildStyles = (mode: string, uiTheme: UITheme) => {
  EStyleSheet.build({
    $theme: mode,
    $androidSummaryFontWeight: uiTheme.androidSummaryFontWeight,
    $link: uiTheme.colors.$link,
    $resolved: uiTheme.colors.$icon,
    $disabled: uiTheme.colors.$disabled,
    $icon: uiTheme.colors.$icon,
    ...uiTheme.colors,
  });
};
export const getThemeMode = async (): Promise<string> => {
  let mode: string;

  try {
    const storedMode: string | null | undefined = JSON.parse(
      await AsyncStorage.getItem(THEME_MODE_KEY),
    );
    mode = storedMode || getSystemThemeMode();
  } catch (e) {
    mode = DEFAULT_THEME.mode;
  }

  return mode;
};
