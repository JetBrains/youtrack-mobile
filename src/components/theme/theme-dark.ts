/* @flow */

import {themeColorsPalette} from './theme-common';

import type {UITheme} from 'flow/Theme';

const dark: UITheme = {
  dark: true,
  mode: 'ytdark',
  name: 'dark',
  barStyle: 'light-content',
  androidSummaryFontWeight: '400',

  colors: {
    $background: '#1c1c1e',
    $boxBackground: '#2c2c2e',
    $dimBackground: 'rgba(0,0,0,0.7)',

    $error: '#C22731',

    $text: '#e1e1e1',
    $textSecondary: '#b0b0b0',
    $textButton: '#FFFFFF',

    $link: '#FF008C',
    $linkLight: 'rgba(254, 0, 130, 0.3)',

    $disabled: '#5d5d5d',

    $icon: '#9f9f9f',
    $iconAccent: '#577996',

    $mask: '#FFFFFF57',
    $navigation: '#696969',

    $separator: 'rgba(210, 254, 254, 0.15)',

    ...themeColorsPalette,
  },
};

export default dark;
