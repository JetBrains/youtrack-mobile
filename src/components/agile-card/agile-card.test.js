import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AgileCard from './agile-card';
import Mocks from '../../../test/mocks';
import {buildStyles} from '../theme/theme';

describe('<AgileCard/>', () => {
  let issueMock;

  beforeAll(() => buildStyles());

  beforeEach(() => {
    Mocks.setStorage({});
    issueMock = Mocks.createIssueMock();
  });

  it('should init', () => {
    const wrapper = shallow(<AgileCard issue={issueMock} />);

    expect(wrapper).toBeDefined();
  });

  it('should render snapshot', () => {
    const tree = shallow(<AgileCard issue={issueMock} />);

    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should show summary', () => {
    const wrapper = shallow(<AgileCard issue={issueMock} />);
    const summary = wrapper.find({testID: 'card-summary'}).children();

    expect(summary.text()).toEqual(issueMock.summary);
  });
});
