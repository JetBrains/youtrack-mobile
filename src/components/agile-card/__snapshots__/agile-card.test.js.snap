// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AgileCard/> should render snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f6f6f6",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginBottom": 8,
        "marginRight": 16,
        "overflow": "hidden",
      },
      undefined,
      undefined,
      undefined,
      Object {
        "height": 110,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderBottomLeftRadius": 8,
          "borderTopLeftRadius": 8,
          "flexShrink": 0,
          "marginBottom": 2,
          "marginTop": 2,
          "width": 4,
        },
        Object {
          "backgroundColor": "initial",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "flexGrow": 1,
          "padding": 14,
        },
        Object {
          "padding": 8,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#737577",
                "flexGrow": 0,
                "flexShrink": 0,
                "fontSize": 14,
                "letterSpacing": -0.17,
              },
              Object {
                "fontSize": 11,
              },
            ]
          }
          testID="card-simple-issue-id"
        >
          00-00
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "flexGrow": 1,
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1f2326",
                "flex": 1,
                "flexGrow": 1,
                "fontSize": 16,
                "fontWeight": "500",
                "letterSpacing": -0.19,
                "lineHeight": 20,
                "marginTop": 8,
              },
              Object {
                "fontSize": 11,
              },
            ]
          }
          testID="card-summary"
        >
          Issue test summary
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
